cli/crates/server/src/bridge/search.rs

```diff
@@ -0,0 +1,401 @@
+        trace!("Indexed {record_count} documents.");
+
+        Ok(Index {
+            inner: index,
+            default_fields,
+            id_field,
+        })
+    }
+}
+
+fn add_field(
+    doc: &mut Document,
+    field: Field,
+    SearchField {
+        name,
+        r#type: base_type,
+    }: &SearchField,
+    RecordDocument { document, .. }: &RecordDocument,
+) -> Result<(), DynamoItemFieldError> {
+    if let Some(value) = document.get(name) {
+        match base_type.as_str() {
+            scalars::URL | scalars::EMAIL | scalars::PHONE | scalars::STRING => {
+                for value in DynamoItemExt::flatten(value) {
+                    doc.add_text(field, DynamoItemExt::to_str(value)?);
+                }
+            }
+            scalars::INT => {
+                for value in DynamoItemExt::flatten(value) {
+                    doc.add_i64(field, DynamoItemExt::to_i64(value)?);
+                }
+            }
+            scalars::FLOAT => {
+                for value in DynamoItemExt::flatten(value) {
+                    doc.add_f64(field, DynamoItemExt::to_f64(value)?);
+                }
+            }
+            scalars::DATE_TIME => {
+                for value in DynamoItemExt::flatten(value) {
+                    doc.add_date(
+                        field,
+                        tantivy::DateTime::from_timestamp_millis(DynamoItemExt::to_datetime(value)?.timestamp_millis()),
+                    );
+                }
+            }
+            scalars::DATE => {
+                for value in DynamoItemExt::flatten(value) {
+                    let value = Utc.from_utc_datetime(
+                        &DynamoItemExt::to_date(value)?.and_time(NaiveTime::from_hms_opt(0, 0, 0).expect("Valid time")),
+                    );
+                    doc.add_date(
+                        field,
+                        tantivy::DateTime::from_timestamp_millis(value.timestamp_millis()),
+                    );
+                }
+            }
+            scalars::TIMESTAMP => {
+                for value in DynamoItemExt::flatten(value) {
+                    doc.add_date(
+                        field,
+                        tantivy::DateTime::from_timestamp_millis(
+                            DynamoItemExt::to_timestamp(value)?.timestamp_millis(),
+                        ),
+                    );
+                }
+            }
+            scalars::BOOL => {
+                for value in DynamoItemExt::flatten(value) {
+                    doc.add_bool(field, DynamoItemExt::to_bool(value)?);
+                }
+            }
+            scalars::IP => {
+                for value in DynamoItemExt::flatten(value) {
+                    doc.add_ip_addr(field, DynamoItemExt::to_ipaddr(value)?);
+                }
+            }
+            _ => unreachable!("Field could not have been added to the schema."),
+        };
+    }
+    Ok(())
+}
+
+#[derive(Debug)]
+pub enum DynamoItemFieldError {
```

Comment:
Not sure we should explicitly name these Dynamo, the datastore being Dynamo compatible is for convenience

Reply:
I've changed the error & result name, but I'd keep the `DynamoItemExt` name. Whether by convenience or by necessity, the documents are constructed as DynamoDB items.